pkg/reconciler/nodeconfig_reconciler.go

```diff
@@ -61,95 +68,208 @@ func (ncr *NodeConfigReconciler) reconcileDebounced(ctx context.Context) error {
 		return fmt.Errorf("error listing revisions: %w", err)
 	}
 
-	var revisionToDeploy *v1alpha1.NetworkConfigRevision
-	for i := range revisions.Items {
-		if revisions.Items[i].Status.IsInvalid {
-			continue
-		}
-		revisionToDeploy = &revisions.Items[i]
-		break
-	}
+	revisionToDeploy := getFirstValidRevision(revisions.Items)
 
 	// there is nothing to deploy - skip
 	if revisionToDeploy == nil {
 		return nil
 	}
 
 	nodes, err := listNodes(ctx, ncr.client)
-
 	if err != nil {
 		return fmt.Errorf("error listing nodes: %w", err)
 	}
 
-	if err := ncr.deployNodeConfigs(ctx, nodes, revisionToDeploy); err != nil {
-		return fmt.Errorf("error deploying node configurations: %w", err)
+	nodeConfigs, err := ncr.listConfigs(ctx)
+	if err != nil {
+		return fmt.Errorf("error listing configs: %w", err)
+	}
+
+	available := len(nodes)
+	ready := 0
+	ongoing := 0
```

Comment:
can we return these instead of passing references?

Reply:
Done